test/test_model.py

```diff
@@ -65,202 +71,230 @@ def setUp(self):
             secretfoo:
                 type: secret
         ''')
-        self.addCleanup(self.harness.cleanup)
-        self.relation_id_db0 = self.harness.add_relation('db0', 'db')
-        self.harness._get_backend_calls(reset=True)
-        self.model = self.harness.model
+        yield harness
+        harness.cleanup()
+
+    @pytest.fixture
+    def model(self, harness: ops.testing.Harness[ops.CharmBase]):
+        return harness.model
+
+    @pytest.fixture
+    def relation_id_db0(self, harness: ops.testing.Harness[ops.CharmBase]):
+        rel_id = harness.add_relation('db0', 'db')
+        harness._get_backend_calls(reset=True)
+        return rel_id
 
     def ensure_relation(
             self,
+            model: ops.Model,
             name: str = 'db1',
             relation_id: typing.Optional[int] = None) -> ops.Relation:
-        """Wrapper around self.model.get_relation that enforces that None is not returned."""
-        rel_db1 = self.model.get_relation(name, relation_id)
+        """Wrapper around model.get_relation that enforces that None is not returned."""
+        rel_db1 = model.get_relation(name, relation_id)
         assert rel_db1 is not None
         assert rel_db1 is not None  # Type checkers understand this, but not the previous line.
         return rel_db1
 
-    def test_model_attributes(self):
-        assert self.model.app is self.model.unit.app
-        assert self.model.name is None
+    def test_model_attributes(self, model: ops.Model):
+        assert model.app is model.unit.app
+        assert model.name is None
```

Comment:
Yeah, I think this would be simpler to understand without the nested harness > model fixture, and just use a harness fixture:
```suggestion
    def test_model_attributes(self, harness):
        assert harness.model.app is harness.model.unit.app
        assert harness.model.name is None
```